rotary/rotary.py

```diff
@@ -136,6 +139,41 @@ def init(args):
     dataset.create_sample_tsv(output_dir_path, header=sample_tsv_header_fields)
 
 
+def download(args):
+    """
+    Runs code for the download command-line mode. Downloads database files required by rotary.
+
+    :param args: The command-line arguments.
+    """
+    config_path = get_config_path(args, default_config_path=provided_config_path)
+    config = load_yaml_config(config_path)
+
+    # modify config to use available CPU and RAM.
+    config = modify_config_with_available_computational_resources(config)
+
+    output_dir_path = os.getcwd()
+
+    config = setup_database_dir(args, config, output_dir_path)
+
+    jobs = get_cli_arg(args, 'jobs')
+    snakemake_args = get_snakemake_args(args)
+
+    # Tell snakemake to only run the download rule.
+    snakemake_args.append('download')
```

Comment:
I just want to check -- do you also need to add `--until` as an arg before `download` for this to work? Thanks.

Reply:
@jmtsuji In Snakemake, you can add a single rule at the very end of the command, and it will cherrypick out that rule, run only that rule, and then evaluate the DAG based on that rule's required inputs. That's how you run Snakemake if you don't have an `all` rule that gets evaluated automatically. So you could have a collection of rules in a single file and call them individually. If you look at the snakemake CLI page, you will note that at the end of the command, there is an option for `targets` (https://snakemake.readthedocs.io/en/stable/executing/cli.html). By putting the rule 'download' at the end of the command, I'm telling Snakemake to target only that rule.

When you use `--until,` it evaluates the complete DAG and runs Snakemake until it hits that rule. When I used `--until,` the code failed during DAG evaluation due to missing certain files and config parameters; you skipped the full DAG evaluation by telling it to target the `download` rule and evaluate a DAG from there. This causes you to only evaluate the download rules, which require no inputs other than a download directory.

Reply:
Got it -- thanks! Yes, this makes sense. I remember this snakemake syntax now.